service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/bucket/SetBucketRolesStep.java

```diff
@@ -4,93 +4,54 @@
-import bio.terra.workspace.service.workspace.GcpCloudContextService;
 import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
-import com.google.api.services.storagetransfer.v1.Storagetransfer;
-import java.io.IOException;
-import java.util.List;
-import java.util.UUID;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 /**
  * Give the Storage Transfer Service SA the appropriate roles on the source and destination (sink)
  * buckets to allow a transfer job to be created.
  *
- * <p>Preconditions: Destination bucket is created. Working map contains DESTINATION_WORKSPACE_ID.
+ * <p>Preconditions: Destination bucket is created.
  *
  * <p>Post conditions: Working map updated with SOURCE_CLONE_INPUTS, DESTINATION_CLONE_INPUTS,
  * CONTROL_PLANE_PROJECT_ID, and STORAGE_TRANSFER_SERVICE_SA_EMAIL. IAM roles are added to both the
  * source and destination buckets in GCS that will allow the transfer service SA the necessary
  * access to create a Storage Transfer Service Job.
  */
 public class SetBucketRolesStep implements Step {
-  private static final List<String> DESTINATION_BUCKET_ROLE_NAMES =
-      Stream.of("roles/storage.legacyBucketWriter").collect(Collectors.toList());
-  private static final List<String> SOURCE_BUCKET_ROLE_NAMES =
-      Stream.of("roles/storage.objectViewer", "roles/storage.legacyBucketReader")
-          .collect(Collectors.toList());
 
-  private final ControlledGcsBucketResource sourceBucket;
-  private final GcpCloudContextService gcpCloudContextService;
   private final BucketCloneRolesService bucketCloneRolesService;
-  private final Storagetransfer storagetransfer;
 
-  public SetBucketRolesStep(
-      ControlledGcsBucketResource sourceBucket,
-      GcpCloudContextService gcpCloudContextService,
-      BucketCloneRolesService bucketCloneRolesService,
-      Storagetransfer storagetransfer) {
-    this.sourceBucket = sourceBucket;
-    this.gcpCloudContextService = gcpCloudContextService;
+  public SetBucketRolesStep(BucketCloneRolesService bucketCloneRolesService) {
     this.bucketCloneRolesService = bucketCloneRolesService;
-    this.storagetransfer = storagetransfer;
   }
 
   @Override
   public StepResult doStep(FlightContext flightContext)
       throws InterruptedException, RetryException {
-    final FlightMap workingMap = flightContext.getWorkingMap();
+    FlightMap workingMap = flightContext.getWorkingMap();
+    // Validate the outputs, so we fail fast if one goes missing.
     FlightUtils.validateRequiredEntries(
-        flightContext.getInputParameters(), ControlledResourceKeys.DESTINATION_WORKSPACE_ID);
-
-    // Gather bucket inputs and store them in the working map for the next
-    // step.
-    final BucketCloneInputs sourceInputs = getSourceInputs();
-    workingMap.put(ControlledResourceKeys.SOURCE_CLONE_INPUTS, sourceInputs);
+        workingMap,
```

Comment:
Instead of doing the validation,  I prefer retrieving data like:
```
var storageTransferServiceSAEmail =
  FlightUtils.getRequired(workingMap, ControlledResourceKeys.STORAGE_TRANSFER_SERVICE_SA_EMAIL, String.class);
```
The advantage is, then IntelliJ and Sonar know that storageTransferServiceSAEmail cannot be null, so they don't complain.